civicrm/civicrm-wordpress
# Add utility method to return a Base Page URL regardless of context

Overview
----------------------------------------
Adds a utility method to return a Base Page URL regardless of context. Fixes [this issue on Lab](https://lab.civicrm.org/dev/wordpress/-/issues/144).

Steps to reproduce
----------------------------------------

1. Create a function:

```php
function civicrm_test_foo() {
  if (!civi_wp()->initialize()) {
    return '';
  }
  $url = CRM_Utils_System::url("civicrm/event/foo", "reset=1", TRUE, NULL, TRUE);
  return $url;
}
```

2. Create a page template, e.g. `page-foo.php` and a WordPress Page that uses the template. 
3. Add a CiviCRM Shortcode to the Page content.
4. In the page template, add:

```php
<pre><?php echo civicrm_test_foo(); ?></pre>
```

You should see:

```
https://example.org/foo/?civiwp=CiviCRM&q=civicrm%2Fevent%2Ffoo&reset=1
```

This is because CiviCRM is set to "Shortcode Mode" by the presence of the Shortcode in the content.

After
----------------------------------------
1. Update the function:

```php
function civicrm_test_foo() {
  // You can use "civicrm_basepage_url()" if you prefer.
  $url = civi_wp()->basepage->url("civicrm/event/foo", "reset=1", TRUE, NULL, TRUE);
  return $url;
}
```

You should see:

```
https://example.org/civicrm/event/foo/?reset=1
```

If you have Polylang installed, you should see the link point to the appropriate Base Page for the language of the Page. When you're viewing a translated page (e.g. in German), then you should see:

```
https://example.org/de/civicrm/event/foo/?reset=1
```

Comments
----------------------------------------

This PR also cleans up another "URL getter" method `CiviCRM_For_WordPress::get_base_url()` because it's no longer necessary. Adds a deprecation notice for any external code that calls it.
